src/odemis/driver/orsay.py

```diff
@@ -2648,30 +2682,336 @@ class Scanner(model.Emitter):
+        """
+        for c in preset:
+            # Only interested in presets which actually defined the right component
+            if c.get("Name") == sub_comp:
+                for s in c:
+                    if s.get("Name") == setting:
+                        for t in s:
+                            if t.tag == tag:
+                                return t.text
+        else:
+            logging.warning(f"Did not find any value for the preset {preset}, with component {sub_comp} "
+                            f"and the setting {setting} using the tag {tag}.")
+            return None
+
+    def _getApertureNmbrFromPreset(self, preset):
+        """
+        Get the aperture number from a given preset.
+
+        :param preset (xml.etree.ElementTree.Element): Full preset XML ElementTree including sub elements with all sub
+        components
+        :return (str): Value found for the setting
+        :raises LookupError if no condenser voltage is found for a preset
+        """
+        aperture_number = self.getPresetSetting(preset, 'HybridAperture', "SelectedDiaph", tag="Target")
+        if aperture_number:
+            return int(aperture_number)
+        else:
+            raise LookupError(f"No aperture number preset found, None type was returned.")
+
+    def _getCondenserVoltageFromPreset(self, preset):
+        """
+        Get the condenser voltage from a given preset.
+
+        :param preset (xml.etree.ElementTree.Element): Full preset XML ElementTree including sub elements with all sub
+        components
+        :return (str): Value found for the setting
+        :raises LookupError if no condenser voltage is found for a preset
+        """
+        condenser_voltage = self.getPresetSetting(preset, 'HVPSFloatingIon', "CondensorVoltage", tag="Target")
+        if condenser_voltage:
+            return float(condenser_voltage)
+        else:
+            raise LookupError(f"No condenser voltage preset found, None type was returned.")
+
+    def _listenerImageFormat(self, image_format):
+        scale = (self.shape[0] / image_format[0],
+                self.shape[1] / image_format[1])
+        if scale != self.scale.value:
+            self.scale._value = scale  # Don't call the setter
+            self.scale.notify(scale)  # Do inform all the subscribers.
+
+    def _setScale(self, scale):
+        self.parent._fib_beam.imageFormat.value = (self.shape[0] / scale[0],
+                                                   self.shape[1] / scale[1])
+        return scale
+
+    def _updatePixelSize(self, *args):
+        """
+        Updates the pixel size VA and the pixel size metadata
+        """
+        self.pixelSize._value = (self.scale.value[0] * (self._fib_beam.horizontalFov.value/self.shape[0]),
+                                self.scale.value[1] * (self._fib_beam.horizontalFov.value / self.shape[1]))
+        self._metadata[MD_PIXEL_SIZE] = self.pixelSize.value
```

Comment:
Confusingly, `.pixelSize` and MD_PIXEL_SIZE don't contain the same value. `.pixelSize` is always computed as if scale = 1. 
Also, to update `.pixelSize`, use `self.pixelSize._set_value(pxs, force_write=True)`. This ensures that the subscribers are notified of changes.